### PR TITLE
Prepare Thunder for Drupal 8.6

### PR DESCRIPTION
**Pull request by chrfritsch**
Make sure these boxes are checked before submitting your pull request - thank you!

- [x] All coding styles are fulfilled. ([How to check for cs issues?](https://github.com/BurdaMagazinOrg/thunder-dev-tools/blob/master/README.md#code-style-guidelines))
- [x] All tests are running locally. ([How to run the test?](https://github.com/BurdaMagazinOrg/thunder-distribution/blob/develop/docs/development.md#how-to-run-the-tests))
- [x] Necessary update hooks are provided.
- [x] User roles have correct access for new introduced permission.
- [x] Every thunder module has a README.md in its root. Follow [this guidelines](https://www.drupal.org/node/2181737), but we don't need every topic.
- [x] Code is covered with well-balanced amount of inline comments.

If you are really awesome, then your feature is covered by additional tests. Well done!

### CODE REVIEW

File: modules/thunder_taxonomy/thunder_taxonomy.install [hunk 1]
```diff
@@ -87,3 +70,29 @@ function thunder_taxonomy_update_8002() {
...
+  $entity_type->set('entity_keys', $entity_keys);
+
+  $definition_update_manager->updateEntityType($entity_type);
+
+  /** @var \Drupal\Core\Entity\EntityFieldManagerInterface $field_manager */
+  $field_manager = \Drupal::service('entity_field.manager');
+  $status = $field_manager->getBaseFieldDefinitions('taxonomy_term')['status'];
+
+  $definition_update_manager->updateFieldStorageDefinition($status->getFieldStorageDefinition());
+
```
Comment (alexpott) [inline modules/thunder_taxonomy/thunder_taxonomy.install:95]: I think we need to do something like
`
  // Uninstall the 'content_translation_status' field if needed.
  if ($has_content_translation_status_field) {
    $content_translation_status = $definition_update_manager->getFieldStorageDefinition('content_translation_status', 'taxonomy_term');
    $definition_update_manager->uninstallFieldStorageDefinition($content_translation_status);
  }
`
ala https://www.drupal.org/node/2985366

File: modules/thunder_taxonomy/thunder_taxonomy.install [hunk 2]
```diff
@@ -87,3 +70,29 @@ function thunder_taxonomy_update_8002() {
   // Output logged messages to related channel of update execution.
   return $thunderUpdater->logger()->output();
 }
+
+/**
+ * Use core's term status field definition.
+ */
+function thunder_taxonomy_update_8003() {
```
Comment (alexpott) [inline modules/thunder_taxonomy/thunder_taxonomy.install:77]: We have a problem with taxonomy_update_8601(). Our users will get a message which if they followed the instructions would result in additional unnecessary fields on their taxonomy. See https://www.drupal.org/project/drupal/issues/2991346

File: thunder.install [hunk 3]
```diff
@@ -66,6 +66,10 @@ function thunder_update_dependencies() {
     $info[$updates[$module]['start']] = $list;
   }
 
+  // Set a custom skip message for update taxonomy_update_8601(), because we
+  // provide our own update path in thunder_taxonomy_update_8003().
+  \Drupal::state()->set('taxonomy_update_8601_skip_message', 'This update function was skipped, because Thunder provides it\'s own update for the status field on taxonomy terms.');
```
Comment (alexpott) [inline thunder.install]: Here we can be smart and only set this if the taxonomy_update_8601 is in the list of pending updates.

**Review by alexpott — CHANGES_REQUESTED** [review]
Just one thing left from my side. Everything else looks really good.

**Review by alexpott — APPROVED** [review]

**Comment by chrfritsch** [comment]
This is not yet ready. The Drupal\Tests\thunder\Functional\Installer\ThunderInstallerTest is still failing for some reason on php5.6
This test is only executed if "[TEST_UPDATE=true]" is in the commit message.

File: composer.json [hunk 4]
```diff
@@ -70,6 +70,9 @@
             },
             "drupal/entity_browser": {
                 "Provide inline entity form FieldWidgetDisplay": "https://www.drupal.org/files/issues/2858438_6.patch"
+            },
+            "drupal/core": {
+                "The message displayed by taxonomy_update_8601() should be overridable": "https://www.drupal.org/files/issues/2018-08-08/2991346-4.patch"
```
Comment (alexpott) [inline composer.json]: This made it in so we no longer need it.

**Review by alexpott — DISMISSED** [review]
Need to remove the patch

**Comment by chrfritsch** [comment]
Looks like we have to adjust the update path again
https://www.drupal.org/project/drupal/issues/2991902

**Comment by chrfritsch** [comment]
I think this is ready now

**Review by dbosen — APPROVED** [review]
So nice!

**Pull request merged**